File: app/src/main/java/com/unique_upgrade/proHeal/OTPActivity.java
package com.unique_upgrade.proHeal;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.preference.PreferenceManager;
import android.text.TextUtils;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class OTPActivity extends AppCompatActivity {
EditText otp;
Button submit_otp;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_o_t_p);
        otp=findViewById(R.id.otp);
        submit_otp=findViewById(R.id.submit_otp);

        submit_otp.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                VerifyOTP();
            }
        });



    }

    private void VerifyOTP() {
        String Otp_text=otp.getText().toString();
        if(Otp_text.equals("0909")){

            getSharedPreferences("ProHeal",MODE_PRIVATE).edit().putString("key","login").commit();
            Intent i=new Intent(OTPActivity.this,MainActivity.class);
            i.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK|Intent.FLAG_ACTIVITY_CLEAR_TASK);
            startActivity(i);


        }
        else{
            Toast.makeText(this, "wrong OTP !!!", Toast.LENGTH_SHORT).show();
        }
    }


}

File: app/src/main/java/com/unique_upgrade/ReliefPro/OTPActivity.kt
package com.unique_upgrade.ReliefPro

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.unique_upgrade.ReliefPro.OTPActivity

class OTPActivity : AppCompatActivity() {
    var otp: EditText? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_o_t_p)
        var submit_otp: Button? = null
        otp = findViewById(R.id.otp)
        submit_otp = findViewById(R.id.submit_otp)
        submit_otp.setOnClickListener(View.OnClickListener { VerifyOTP() })
    }

    private fun VerifyOTP() {
        val Otp_text = otp!!.text.toString()
        if (Otp_text == "0909") {
            getSharedPreferences("ReliefPro", MODE_PRIVATE).edit().putString("key", "login").commit()
            val i = Intent(this@OTPActivity, MainActivity::class.java)
            i.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK)
            startActivity(i)
        } else {
            Toast.makeText(this, "wrong OTP !!!", Toast.LENGTH_SHORT).show()
        }
    }
}